sdk/core/azure_core/src/http/options/mod.rs

```diff
@@ -41,6 +43,9 @@ pub struct ClientOptions {
     ///
     /// Specifies which headers and query parameters should be logged. All headers and query parameters not in the allow list will be redacted.
     pub logging: LoggingOptions,
+
+    /// Cloud configuration for the client. If None, the client will default to Azure Public Cloud.
+    pub cloud: Option<Arc<CloudConfiguration>>,
```

Comment:
This isn't the current pattern: any options that are options bags themselves are just inline - not `Option<T>`. Only leave nodes are `Option<T>` unless they also have a zero-size default like `Vec` and `HashMap`.

Reply:
...though, as I think about this more, maybe it does make sense here since it'd just ben an `Arc<CloudConfiguration>` otherwise, which we don't often need. You know what? Never mind on this one. I think it makes sense as is but saving this thread for posterity.